### PR TITLE
Feature/chrome extension

### PR DESCRIPTION
**Pull request by gamalielhere**
### Feature

- [x] Updated CHANGELOG.md
- [x] Add PR label

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -4,13 +4,15 @@ GITHUB_TOKEN="$2"
...
+cd dist; zip -r ../release/MyEtherWallet-$RELEASE.zip *; sha256sum * > ../release/MyEtherWallet-$RELEASE-CHECKSUM-SHA256; cd ..
+cd chrome-extension; zip -r ../release/MEW-CX-$RELEASE.zip *; sha256sum * >  ../release/MEW-CX-$RELEASE-CHECKSUM-SHA256; cd ..
+
 if [ -n "$4" ]; then
     for f in release/*; do
         gpg --output $f.sig --detach-sig $f
     done
 fi
-cd release && sha256sum * > MyEtherWallet-$RELEASE-CHECKSUM-SHA256 && cd ..
+
```
Comment (kvhnuke) [inline release.sh]: why was this removed?

File: src/builds/mewcx/cxHelpers/background.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+      return dom === extractRootDomain(tab.url);
+    });
+
+    if (foundWhitelist === undefined) {
+      if (
+        foundBlacklist !== undefined ||
+        helpers.checkUrlSimilarity(tab.url, SEARCH_STRING)
+      ) {
+        urlRedirect = encodeURI(
+          `https://www.myetherwallet.com/phishing.html?phishing-address=${tab.url}`
```
Comment (kvhnuke) [inline src/builds/mewcx/cxHelpers/background.js]: shouldn't this be `phishing-catcher?phishing-address`

File: src/containers/HeaderContainer/HeaderContainer.vue [hunk 3]
```diff
@@ -78,14 +85,13 @@
...
+                <b-nav-item v-if="isHomePage" to="/" exact>{{
+                  $t('header.home')
+                }}</b-nav-item>
                 <b-nav-item
-                  v-if="isHomePage"
-                  to="/"
-                  exact
-                  @click="scrollTop()"
-                  >{{ $t('header.home') }}</b-nav-item
+                  v-if="isHomePage && buildType !== 'mewcx'"
```
Comment (kvhnuke) [inline src/containers/HeaderContainer/HeaderContainer.vue]: can we import the buildType `MEW_CX` without hardcoding it

File: src/helpers/ExtensionHelpers.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+const getAccounts = callback => {
+  const chrome = window.chrome;
+  chrome.storage.sync.get(null, callback);
+};
+
+const getPrivFromMnemonicWallet = (mnem, path) => {
+  const seed = bip39.mnemonicToSeedSync(mnem);
+  const hdKey = HDKey.fromMasterSeed(seed);
+  return hdKey.derive(path).privateKey;
+};
```
Comment (kvhnuke) [inline src/helpers/ExtensionHelpers.js]: you can use `import MnemonicWallet from '@/wallets'`, it will probably be better since it opens up other functionalities as well

File: src/helpers/ExtensionHelpers.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        return;
+      }
+    } else {
+      if (foundAddress) {
+        Toast.responseHandler('Address already stored!', Toast.ERROR);
+        return;
+      }
+    }
+
+    nickname = nickname.replace(/(<([^>]+)>)/gi, '');
```
Comment (kvhnuke) [inline src/helpers/ExtensionHelpers.js]: do a xss strip

File: src/helpers/createBlob.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export default (str, mime) => {
+  const string = typeof str === 'object' ? JSON.stringify(str) : str;
+  if (string === null) return '';
+  const blob = new Blob([string], {
+    type: mime
+  });
+  return window.URL.createObjectURL(blob);
+};
```
Comment (kvhnuke) [inline src/helpers/createBlob.js]: can you check how we do it for web file download and may be use the same function if it is similiar?

File: src/layouts/AccessWalletLayout/components/Web3WalletModal/Web3WalletModal.vue [hunk 7]
```diff
@@ -98,12 +118,16 @@
...
+              isMetaMask
+                ? 'https://metamask.io/'
+                : 'https://chrome.google.com/webstore/detail/myetherwallet/nlbmnnijcnlegkjjpcfjclmcfggfefdm?hl=en'
+            "
             target="_blank"
             rel="noopener noreferrer"
             class="mid-round-button-green-filled close-button"
             @click="refreshPage = true"
-            >{{ $t('accessWallet.installMetamask') }}</a
+            >{{ isMetaMask ? 'Install MetaMask' : 'Install Web3 Wallet' }}</a
```
Comment (kvhnuke) [inline src/layouts/AccessWalletLayout/components/Web3WalletModal/Web3WalletModal.vue]: this should be our Chrome extension link

File: src/layouts/ExtensionBrowserAction/components/WalletInfoComponent/WalletInfoComponent.vue [hunk 8]
```diff
@@ -0,0 +1,571 @@
...
+        tokens = tokens.map(token => {
+          const balance = token.balance;
+          delete token.balance;
+          token.balance = new BigNumber(balance).gt(0)
+            ? new BigNumber(balance)
+                .div(new BigNumber(10).pow(token.decimals))
+                .toFixed(3)
+            : 0;
+          token.address = token.addr;
+          // token['logo'] = token.hasOwnProperty('logo') ? token.logo.hasOwnProperty('src') ? token.logo : newLogo : newLogo
```
Comment (kvhnuke) [inline src/layouts/ExtensionBrowserAction/components/WalletInfoComponent/WalletInfoComponent.vue]: comment

File: src/store/state.js [hunk 9]
```diff
@@ -1,14 +1,13 @@
...
-import darklist from '@/darklist/address-darklist.json';
+import darklist from '@/address-darklist/address-darklist.json';
 import store from 'store';
 if (store.get('notifications') === undefined) store.set('notifications', {});
-
 const gettingStartedDone =
   store.get('skipTutorial') !== undefined ? store.get('skipTutorial') : false;
 const storedNetwork = store.get('network');
-let network = nodeList['ETH'][0];
+let network = BUILD_TYPE !== 'mewcx' ? nodeList['ETH'][2] : nodeList['ETH'][1];
```
Comment (kvhnuke) [inline src/store/state.js]: fallback should be `nodeList['ETH'][0]`

File: src/store/state.js [hunk 10]
```diff
@@ -26,12 +25,13 @@ const state = {
...
     isHardware: false,
-    identifier: ''
+    identifier: '',
+    nickname: ''
   },
   customPaths: customPaths,
   ens: null,
   Errors: {},
-  ethDonationAddress: '0xDECAF9CD2367cdbb726E904cD6397eDFcAe6068D',
+  ethDonationAddress: '0xDECAF9CDx2367cdbb726E904cD6397eDFcAe6068D',
```
Comment (kvhnuke) [inline src/store/state.js]: address is wrong has a `x`

File: src/helpers/createBlob.js (hunk 6 above)
Comment (gamalielhere) [reply src/helpers/createBlob.js]: Oh this refactors that. Both the keystore download uses this function now
